internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -351,40 +355,54 @@ func (r *BMACReconciler) handleBMHFinalizer(ctx context.Context, log logrus.Fiel
 
 	// BMH is being deleted and finalizer is gone, ensure detached and paused are gone and return
 	if !bmhHasFinalizer {
-		dirty := removeBMHDetachedAndPausedAnnotations(log, bmh)
+		if removeAgentOnDelete {
+			removeBMHDetachedAnnotation(log, bmh)
+		}
+		if bmhPaused {
```

Comment:
I still don't think you need to be checking this at all.
`removeBMHPausedAnnotation` is already a no-op if there's no annotation present.